problems/problem.go

```diff
@@ -33,7 +34,8 @@ type RequestDecoratableProblem interface {
 // FromError, convert an Go error into a Problem. This is a no-op if the supplied
 // error already is a problem. Otherwise the returned error is an InternalProblem (HTTP 500).
 func FromError(err error) Problem {
-	if problem, alreadyProblem := err.(Problem); alreadyProblem {
+	var problem Problem
+	if errors.As(err, &problem) {
```

Comment:
This would break existing users that uses the following,
* GetA() and GetB() returns errors/problems
* GetA() calls GetB() and if there is a NotFoundProblem it should be an error and wraps it for traceability

When this "error" is handled by WriteProblem it will unwrap the first error and display that one instead of the outer error.

Reply:
Fine, we don't want to break it for existing users. That's fair.

Reply:
@Ekberg 
I want to implement this change even though it breaks the scenario you mentioned. I am of the opinion that using wrapping to convert a problem to a 500 is wrong and should be avoided. 

From what I can see, wrap=500 is the only function that is lost by this change. You still get the traceability since we log the original error and not the converted problem
```
l := log.
  WithTracing(ctx).
  WithClientID(ctx).
  WithUserID(ctx).
  WithError(err).
  WithField("code", statusCode)
``` 
What I want to be able to do:
* Create 404 problem deep in the code. 
Expose it as an `error` and wrap it for traceability as you would any error.
* Have flexibility in how to use problems:
  * Problems can be used as any typed error
  * Problems can be obscured (regular errors)
  
These things are possible, however I would need to do a workaround to prevent WriteResponse to interpret my errors as 500 when they aren't. 
If I want to add traceability to a 404 problem (by wrapping it) I am unable to to so. In fact, all problems except 500 are written without logging the context of where they arose.

Reply:
Lets consider it for v1.0 of this repo? Then we can make the breaking change if we still feel like it?